crates/interpreter/src/parser.rs

```diff
@@ -64,6 +64,10 @@ impl<'m, M: Mode> Parser<'m, M> {
         Ok(result)
     }
 
+    pub fn parse_bytes_reversely(&mut self, delta: i32) -> MResult<&'m [u8], M> {
+        unimplemented!()
```

Comment:
The difficulty is that if `delta` is negative, a similar implementation as the `parse_bytes(&mut self, len: usize)` right above would not work because we don't seem to have access to the part of the `self.data` before the current IP. So we may have to save a copy of the entire `self.data` at the beginning, in order to apply negative `delta_ip`. Is that correct?

Reply:
You shouldn't need such a function. The whole point of the side-table is that we don't need to parse bytes anymore, we can directly jump where needed. I just checked in Miri with strict provenance and it seems it's not an issue to enlarge a slice (if it came from one that used to be as large). However you'll need to use unsafe:

```rust
fn jump(cur: &'m [u8], off: isize) -> &'m [u8] {
    debug_assert!(off <= cur.len() as isize);
    unsafe { core::slice::from_raw_parts(cur.as_ptr().offset(off), (cur.len() as isize - off) as usize) }
}
```

With the `toctou` feature, we could indeed save a copy of the parser at the function entry and use that to check that `0 <= (entry.len() - cur.len()) as isize + off`.